torch/backends/cudnn/__init__.py

```diff
@@ -27,7 +27,19 @@
 
 
 def find_cudnn_windows_lib():
-    proc = Popen(['where', 'cudnn64*.dll'], stdout=PIPE, stderr=PIPE, stdin=PIPE)
+    # Override the default search process
+    # The libary selection will be done in these directories one by one
+    # 1. [Package Root]\Lib 
+    #    That's where our libraries are in, which should be loaded first.
+    # 2. Default directories
+    #    That is stored in the environment variable `PATH`.
+    test_env = os.environ.copy()
+    old_path = test_env['PATH']
+    th_dll_path = os.path.join(os.path.dirname(
+        os.path.dirname(os.path.dirname(__file__))), 'lib')
+    test_env['PATH'] = ';'.join([th_dll_path, old_path])
```

Comment:
I was going to complain that this isn't portable but this is Windows only :>

Reply:
I see this code is substantially duplicated for cuda as well. Can we make a helper function to do this instead?

Reply:
It would logically  make sense to me if cudnn imported cuda. Does that result in a circular import?

Reply:
i'm not looking forward to de-copy/paste unless it's necessary because this code is very specific and possibly will change with cudnn / cuda versions.

Reply:
One is in `torch/cuda/__init__.py` and the other one is in `torch/backends/cudnn/__init__.py`. And currently, there is no dependency between these two scripts. A safe and easy option is to put the function inside one file under the root directory of torch.

Reply:
let's just leave them as-is for now

Reply:
@soumith Sure, I'm okay with this.